src/libostree/ostree-sign-dummy.c

```diff
@@ -0,0 +1,187 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+#include "config.h"
+
+#include "ostree-sign-dummy.h"
+#include <string.h>
+
+#undef G_LOG_DOMAIN
+#define G_LOG_DOMAIN "OSTreeSign"
+
+#define OSTREE_SIGN_DUMMY_NAME "dummy"
+
+#define OSTREE_SIGN_METADATA_DUMMY_KEY "ostree.sign.dummy"
+#define OSTREE_SIGN_METADATA_DUMMY_TYPE "aay"
+
+#define OSTREE_SIGN_DUMMY_SIGNATURE "dummysign"
+
+struct _OstreeSignDummy
+{
+  GObject parent;
+  gchar *signature_ascii;
+};
+
+static void
+ostree_sign_dummy_iface_init (OstreeSignInterface *self);
+
+G_DEFINE_TYPE_WITH_CODE (OstreeSignDummy, ostree_sign_dummy, G_TYPE_OBJECT,
+        G_IMPLEMENT_INTERFACE (OSTREE_TYPE_SIGN, ostree_sign_dummy_iface_init));
+
+static void
+ostree_sign_dummy_iface_init (OstreeSignInterface *self)
+{
+  g_debug ("%s enter", __FUNCTION__);
+
+  self->data = ostree_sign_dummy_data;
+  self->get_name = ostree_sign_dummy_get_name;
+  self->metadata_key = ostree_sign_dummy_metadata_key;
+  self->metadata_format = ostree_sign_dummy_metadata_format;
+  self->metadata_verify = ostree_sign_dummy_metadata_verify;
+  self->set_sk = ostree_sign_dummy_set_signature;
+  self->set_pk = ostree_sign_dummy_set_signature;
+}
+
+static void
+ostree_sign_dummy_class_init (OstreeSignDummyClass *self)
+{
+  g_debug ("%s enter", __FUNCTION__);
+}
+
+static void
+ostree_sign_dummy_init (OstreeSignDummy *self)
+{
+  g_debug ("%s enter", __FUNCTION__);
+
+  self->signature_ascii = g_strdup(OSTREE_SIGN_DUMMY_SIGNATURE);
+}
+
+gboolean ostree_sign_dummy_set_signature (OstreeSign *self, GVariant *key, GError **error)
+{
+  g_debug ("%s enter", __FUNCTION__);
+
+  OstreeSignDummy *sign =  ostree_sign_dummy_get_instance_private(OSTREE_SIGN_DUMMY(self));
+
+  if (sign->signature_ascii != NULL)
+    g_free(sign->signature_ascii);
+
+  sign->signature_ascii = g_variant_dup_string (key, 0);
+
+  return TRUE;
+}
+
+gboolean ostree_sign_dummy_data (OstreeSign *self,
+                                 GBytes *data,
+                                 GBytes **signature,
+                                 GCancellable *cancellable,
+                                 GError **error)
+{
+
+  g_debug ("%s enter", __FUNCTION__);
+  g_return_val_if_fail (OSTREE_IS_SIGN (self), FALSE);
+
+  OstreeSignDummy *sign =  ostree_sign_dummy_get_instance_private(OSTREE_SIGN_DUMMY(self));
+
+  *signature = g_bytes_new (sign->signature_ascii, strlen(sign->signature_ascii));
+
+  return TRUE;
+}
+
+gchar * ostree_sign_dummy_get_name (OstreeSign *self)
+{
+  g_debug ("%s enter", __FUNCTION__);
+  g_return_val_if_fail (OSTREE_IS_SIGN (self), FALSE);
+
+  g_autofree gchar *name = g_strdup(OSTREE_SIGN_DUMMY_NAME);
```

Comment:
You are leaking `name` since `ostree_sign_get_name` returns a `const char *`.
Just `return OSTREE_SIGN_DUMMY_NAME` is enough here.